cli/concourseutil.py

```diff
@@ -308,9 +308,11 @@ def trigger_resource_check(
     cfg_set = cfg_factory.cfg_set(cfg_name)
     concourse_cfg = cfg_set.concourse()
     team_credentials = concourse_cfg.team_credentials(team_name)
+    concourse_version = concourse_cfg.concourse_version()
     api = client.ConcourseApi(
```

Comment:
I'd prefer to add a factory function for retrieving a concourse API object of the proper version (and introduce version-specific api-classes). Like so:

```# (in concourse.client.py)
def concourse_api(
  base_url: str,
  team_name: str,
  concourse_version: concourse.client.ConcourseApiVersion
):
 if concourse_version == ConcourseVersion.V3:
  return ConcourseApiV3(...)
elif ...
else: raise NotImplementedError

# maybe rename ConcourseApi -> ConcourseApiBase
class ConcourseApiV3(ConcourseApiBase):
  pass # overwrite any functions that need to be different between V3 and V4

class ConcourseApiV4(ConcourseApiBase):
  pass # overwrite any functions that need to be different between V3 and V4

# same for ApiRoutes

class ConcourseApiVersion(enum.Enum): # --> probably this should go do model/concourse.py (and be used there; i.e. the "version-getter" should already return ConcourseApiVersion instances
 V3 = '3'
 V4 = '4'
```

Implementation note: both Api classes must expose the same methods with the same external behaviour. That way, after creating a ConcourseAPI object, no subsequent for any user of said concourse api instances ougth to be required.

Reply:
Done